lib/tapioca/dsl/compiler.rb

```diff
@@ -21,26 +21,26 @@ class Compiler
       #: Hash[String, untyped]
       attr_reader :options
 
-      @@requested_constants = [] #: Array[Module] # rubocop:disable Style/ClassVars
+      @@requested_constants = [] #: Array[T::Module[top]] # rubocop:disable Style/ClassVars
 
       class << self
         extend T::Sig
 
-        #: (Module constant) -> bool
+        #: (T::Module[top] constant) -> bool
         def handles?(constant)
           processable_constants.include?(constant)
         end
 
         # @abstract
-        #: -> T::Enumerable[Module]
+        #: -> T::Enumerable[T::Module[top]]
         def gather_constants = raise NotImplementedError, "Abstract method called"
 
-        #: -> Set[Module]
+        #: -> Set[T::Module[top]]
         def processable_constants
-          @processable_constants ||= T::Set[Module].new.compare_by_identity.merge(gather_constants) #: Set[Module]?
+          @processable_constants ||= T::Set[T::Module[T.anything]].new.compare_by_identity.merge(gather_constants) #: Set[T::Module[top]]?
```

Comment:
Do we even need the `T::Set` here?
```suggestion
          @processable_constants ||= Set.new.compare_by_identity.merge(gather_constants) #: Set[T::Module[top]]?
```